netgear/netgear.go

```diff
@@ -0,0 +1,174 @@
+package netgear
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const timeout = 10 * time.Second
+
+type Netgear struct {
+	sessionToken string
+	client       *http.Client
+
+	baseUrl  *url.URL
+	username string
+	password string
+}
+
+func NewNetgear(hostName, username, password string) (*Netgear, error) {
+	u, err := url.Parse(strings.TrimSpace(hostName))
```

Comment:
https://github.com/Icinga/check-netgear/pull/4#discussion_r2481808159

```suggestion
func NewNetgear(baseUrl, username, password string) (*Netgear, error) {
	u, err := url.Parse(strings.TrimSpace(baseUrl))
```